### ISSUE TITLE
Running httpserver_test results in a popup on windows and intermittent failures in the CI

### ISSUE THREAD

**Issue by shashankram** [openservicemesh/osm#1562]
**Bug description**:
Running `go test ./pkg/httpserver/` results in a popup on windows and intermittent failures in the CI.
```
{"level":"fatal","component":"http-server","error":"http: Server closed","time":"2020-08-17T16:37:36-07:00","file":"httpserver.go:54","message":"Failed to start API server"}
FAIL	github.com/openservicemesh/osm/pkg/httpserver	0.081s
```

**Affected area (please mark with X where applicable)**:
- Install                    [ ]
- SMI Traffic Access Policy  [ ]
- SMI Traffic Specs Policy   [ ]
- SMI Traffic Split Policy   [ ]
- Permissive Traffic Policy  [ ]
- Ingress                    [ ]
- Egress                     [ ]
- Envoy Control Plane        [ ]
- CLI Tool                   [ ]
- Metrics                    [ ]
- Certificate Management     [ ]
- Sidecar Injection          [ ]
- Logging                    [ ]
- Debugging                  [ ]
- Tests                      [X]
- CI System                  [ ]

**Expected behavior**:
The test should not fail and should also not result in popups when run locally. 

**Steps to reproduce the bug (as precisely as possible)**:
`go test ./pkg/httpserver/`

**Comment by shashankram** [comment]
Seeing more failures:
https://pipelines.actions.githubusercontent.com/FqYyUlFBwgcC4xlcAlRHSOZjGFqxSg5UdT6unwZ3oA6NkQIoht/_apis/pipelines/1/runs/6285/signedlogcontent/43?urlExpires=2020-08-18T00%3A45%3A55.5640407Z&urlSigningMethod=HMACV1&urlSignature=N4hu5AiF%2Fh9mlgzVZCDq3EOeu0uTWnX%2Fb8HNNHD2PSI%3D

```
2020-08-18T00:43:11.5850079Z [1m[32mSUCCESS![0m -- [32m[1m6 Passed[0m | [91m[1m0 Failed[0m | [33m[1m0 Pending[0m | [36m[1m0 Skipped[0m
2020-08-18T00:43:11.5850636Z {"level":"fatal","component":"http-server","error":"http: Server closed","time":"2020-08-18T00:42:31Z","file":"httpserver.go:54","message":"Failed to start API server"}
```

**Comment by SanyaKochhar** [comment]
Hey Shashank, how come this issue needed to be reopened? Will PR #1572 solve this?

**Comment by shashankram** [comment]
> Hey Shashank, how come this issue needed to be reopened? Will PR #1572 solve this?

There are 2 issues:
1. test intermittently failing - resolved by #1565 
2. test resulting in popups in windows env - being resolved by #1572 

Since 1 above was merged, github automatically closed the issue because the PR linked this issue.

Let me know if you have any concerns @SanyaKochhar.
